### DISCUSSION TITLE
نحوه استفاده از Warp+ در مرزبان بدون مشکل مصرف منابع

### DISCUSSION THREAD

**Discussion by M03ED** [Gozargah/Marzban#212] [Show and tell]
با سلام
با توجه به اینکه استفاده از آموزش قبلی برای راه اندازی Warp باعث مصرف بالای منابع سرور میشه میخواهم روش بهتری رو برای اینکار بهتون یاد بدم.

ابتدا برای اینکه بتونیم سرعت بهتری از warp بگیریم از ربات [Warp+ Bot](https://t.me/generatewarpplusbot) یک اکانت Warp+ دریافت می کنیم.

![image](https://user-images.githubusercontent.com/50927468/233935889-02b11b3e-7c76-417a-aab5-c2d1f8eef7c8.png)

حالا باید با استفاده از پروژه [wgcf](https://github.com/ViRb3/wgcf) اون رو تبدیل به فایل وایرگارد کنیم

با استفاده از این 2 دستور پروژه رو ران و یک اکانت موقت میسازیم 

**wgcf register**
**wgcf generate**

![image](https://user-images.githubusercontent.com/50927468/233941594-fccb11dc-b203-4857-a286-1c769a968cac.png)

حالا باید key دریافتی از ربات رو جایگزین licence_key توی فایل wgcf-account کنیم

![image](https://user-images.githubusercontent.com/50927468/233942186-cd2db4e7-964c-4c12-9f6d-7514ca46ebcc.png)

اینبار با دستور **wgcf update**  کانفیگ رو اپدیت می کنیم و می بینیم که حجم ما 21PiB هست 

![image](https://user-images.githubusercontent.com/50927468/233942819-ba78be9e-0bb3-48c2-ba13-43e5eeb09f35.png)

یک بار دیگه دستور **wgcf generate** انجام بدید تا کانفیگ جدید ساخته بشه 

![image](https://user-images.githubusercontent.com/50927468/233943254-30ecfdfb-17b2-4557-b3a3-7d44bdbd26fe.png)

نکته : این مراحل رو میتونید روی کامپیوتر هم انجام بدید و نیازی نداره حتما رو سرور انجام بشه

حالا یک outbound به ضورت زیر به فایل xray اضافه می کنیم 

        {
            "tag":"warp",
            "protocol":"wireguard",
            "settings":{
                "secretKey":"KPbr+HXWocMVfNiF9mkj8USEKJ8c8H6MbE3ULh7eG0w=",
                "DNS" : "1.1.1.1",
                "address":[
                    "172.16.0.2/32",
                    "2606:4700:110:8756:9135:af04:3778:40d9/128"
                ],
                "peers":[
                    {
                        "publicKey":"bmXOC+F1FxEMF9dyiK2H5/1SUtzH0JuVo51h2wPfgyo=",
                        "endpoint":"engage.cloudflareclient.com:2408"
                    }
                ]
            }
        }

فراموش نکنید پرایوت کی خودتون رو جایگزین کنید.

بخش routing رو هم به صورت زیر تغییر بدید

    "routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "outboundTag": "block",
                "ip": [
                    "geoip:private"
                ]
            },
            {
                "type": "field",
                "port": 53,
                "network": "tcp,udp",
                "outboundTag": "DNS-Internal"
            },
            {
                "type": "field",
                "outboundTag": "block",
                "protocol": [
                    "bittorrent"
                ]
            },
            {
                "outboundTag": "warp",
                "domain": [
                    "regexp:.*\\.ir$",
                    "ext:iran.dat:ir",
                    "ext:iran.dat:other",
                    "geosite:category-ir",
                    "openai.com",
                    "ai.com",
                    "ipinfo.io",
                    "iplocation.net",
                    "google.com",
                    "spotify.com"
                ],
                "type": "field"
            }
        ]
    }

سایت های دلخواه خودتون رو به قسمت وارپ اضافه کنید 

نکته : در صورتی میخواهید ترافیک وبسایت های ایرانی رو هم به وارپ منتقل کنید باید یک سری فایل رو دانلود کرده و در فایل env تغییری ایجاد کنید که در [اینجا](https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-5353492) نوشته شده ، در کد بنده هم این انتقال انجام شده.

**Comment by raishiv98** [comment]
مراحلو دقیق انجام دادم ولی سایتایی ک تو وارپ میزارم کار نمیکنن مشکل چیه

**Comment by tbehzad76** [comment]
لطفا برای نود هم ابدیت کنید

**Reply by M03ED** [reply]
نود همین کانفیگ رو به صورت اتوماتیک دریافت می‌کنه و نیازی به تغییر نداره ، توصیه میکنم از xray ورژن 1.8.3 برای این روش استفاده کنید

**Reply by Ramintn** [reply]
سلام تو مرزبان چطوری xray 1.8.3 بزنیم

**Reply by M03ED** [reply]
https://gozargah.github.io/marzban/
تو داکیومنت اموزش داده شده

**Reply by Ramintn** [reply]
خیلی ممنون من که سایت های ایرانی رو بستم هم روی مین سرور و هم رو‌ نودها،،طبق آموزش بالا میتونم وارپ بزنم؟یا باید تغییرات بدم در کدهای بالا؟

**Reply by M03ED** [reply]
این دو موردی که گفتید ربطی به هم ندارن پس میشه

**Comment by Ramintn** [comment]
سلام اگه بخوایم کل ترافیک از وارپ رد بشه و همچنین آیپی سرور برگرده باید چیکار کنیم؟

**Reply by rickyivan** [reply]
نمیشه. یا خدا یا خرما

**Comment by masterj3y** [comment]
سلام تمام مراحل رو دقیق انجام دادم ولی بازم spotify رو باز نمیکنه

**Reply by M03ED** [reply]
https://gozargah.github.io/marzban/examples/warp

**Comment by M03ED** [comment]
لطفا از داکیومنت خود مرزبان استفاده کنید 

https://gozargah.github.io/marzban/examples/warp

**Comment by nilofari700** [comment]
این نود درصد ملت نمیتونن اجرا کنن فقط پنلشونو خراب میکنند. بکنش ی اسکریپ با اجرا و درخواست پارامتر ها انجام داده بشه.

**Reply by M03ED** [reply]
این discussion مدت هاست که بسته شده ، اینکه شما در اجرای چیزی مشکل دارید دلیل بر اشتباه بودن آموزش نمیشه.
همیشه انتظار نداشته باشید لقمه آماده بهتون بدن و کل راه رو با یک اسکریپت انجام بدید.
از داکیومنت خود مرزبان برای راه اندازی وارپ استفاده کنید.
https://gozargah.github.io/marzban/examples/warp

**Comment by oj1378** [comment]
سلام. من هیچی از وارپ و این چیزا نمی دونم و کامنت ها رو خوندم و هیچی متوجه نشدم. در واقع صفر کیلومترم. می خوام واسه خودم فیلتر شکن بسازم و یا از همین وارپ استفاده کنم. پیشنهادتون برای من چیه ؟

**Reply by oj1378** [reply]
این ایدی تلگرام منه خوشحال میشم اگه در حد توانتون منو راهنمایی کنید.  @oj1378

**Comment by arvinmoradi** [comment]
سلام 
چطوری فقط یک پورت رو از warp عبور بدیم ؟؟!!!

**Comment by w0l4i** [comment]
از این بخش برای چه کاری داره استفاده میشه ؟ :

        {
            "type": "field",
            "port": 53,
            "network": "tcp,udp",
            "outboundTag": "DNS-Internal"
        },
